### PR TITLE
llvm (Low-Level Virtual Machine): update to 18

### PR DESCRIPTION
**Pull request by MingcongBai**
Topic Description
-----------------

Update LLVM to 18.x and rebuild all reverse dependencies.

Package(s) Affected
-------------------

- bcc: 0.30.0-1
- bpftrace: 0.21.1-1
- castxml: 0.6.7-1
- ccls: 0.20240202-1
- clazy: 1.11-3
- gnome-builder: 42.1-6
- kdevelop: 23.08.5-1
- ldc: 1.38.0-1
- liblphobos: 1.38.0-1
- llvm: 18.1.8
- llvm-runtime: 18.1.8
- mesa: 1:24.0.7+dxheaders1.613.1-1
- opencl-clang: 18.1.0-1
- postgresql: 13.15-1
- pyside2: 5.15.14-1
- qtcreator: 13.0.2-1
- rustc: 1:1.79.0-1
- spirv-llvm-translator: 18.1.0
- zig: 0.13.0

Security Update?
----------------

No

Build Order
-----------

```
#buildit llvm:+stage2 llvm:-pkgbreak ldc:+stage2 ldc groups/llvm-rebuilds llvm
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

**Experimental Architectures**

- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

File: app-database/postgresql/spec [hunk 1]
```diff
@@ -1,4 +1,6 @@
 VER=13.13
+REL=1
 SRCS="tbl::https://ftp.postgresql.org/pub/source/v$VER/postgresql-$VER.tar.bz2"
 CHKSUMS="sha256::8af69c2599047a2ad246567d68ec4131aef116954d8c3e469e9789080b37a474"
 CHKUPDATE="anitya::id=5601"
+REL=1
```
Comment (jiegec) [inline app-database/postgresql/spec]: Duplicate REL

File: app-database/postgresql/autobuild/overrides/usr/lib/sysusers.d/postgresql.conf [hunk 2]
```diff
@@ -0,0 +1,2 @@
+g    postgres  90
+u    postgres  90:90 "Postgres Daemon Owner" /var/lib/postgres  /bin/bash
```
Comment (jiegec) [inline app-database/postgresql/autobuild/overrides/usr/lib/sysusers.d/postgresql.conf]: Missing end of file newline

File: app-devel/bcc/autobuild/defines [hunk 3]
```diff
@@ -13,6 +13,9 @@ PKGDES="BPF compiler collection and tools for Linux kernel tracing and instrumen
 
 NOSTATIC=0
 
+# FIXME: cmakeninja fails on very fast (circa 2024) machines (such as our
+# towards-modern-distro buildbot).
+ABTYPE=cmake
```
Comment (jiegec) [inline app-devel/bcc/autobuild/defines]: Please try cmakeninja again?

**Comment by jiegec** [comment]
Remember to lower ENVREQ for riscv64 machines.

**Comment by jiegec** [comment]
Let's continue. Does golang actually depend on llvm?

image

**Comment by jiegec** [comment]
ldc will support llvm 18 in next release: https://github.com/ldc-developers/ldc/releases/tag/v1.38.0-beta1

**Comment by jiegec** [comment]
llvm 18.1.5 is out

**Comment by jiegec** [comment]
rust 1.78.0 is out.

**Comment by jiegec** [comment]
If too much work to be done in one pr, we can add llvm-18 and migrate packages to the new version gradually.

**Comment by jiegec** [comment]
https://github.com/ldc-developers/ldc/releases/tag/v1.38.0 ldc llvm18

**Comment by jiegec** [comment]
Mostly done, waiting for #6097

**Review by eatradish — DISMISSED** [review]

**Comment by prcups** [comment]
lgtm

**Review by chenx97 — APPROVED** [review]

**Review by Cyanoxygen — APPROVED** [review]

**Pull request merged**